Assets/MixedRealityToolkit/_Core/Interfaces/IMixedRealityController.cs

```diff
@@ -31,7 +31,7 @@ public interface IMixedRealityController
         /// <summary>
         /// Mapping definition for this controller, linking the Physical inputs to logical Input System Actions
         /// </summary>
-        Dictionary<DeviceInputType, InteractionMapping> Interactions { get; }
+        Dictionary<DeviceInputType, IInteractionMapping> Interactions { get; }
```

Comment:
Not really, as it would further complicate the lookup.  The reason the DeviceInputType is the key, is so that when a button is pressed, the code takes the input and looks for the mapping for that input.
With it being an enum, it's an ID lookup in any case and as fast as we can make it.

Physical -> SDK -> InputType -> InteractionMapping

Reply:
I still think having the device input type as the key limits us with assigning multiple types here.

Reply:
How does it limit us?  A key is unique, so we can't assign multiples.  Plus the interactions are what are configured for that physical input.  Theres not that many to create an issue.  
Although, yes, we do need more input types as we add more physical controllers with more buttons.

It was suggested we also have a filter on the controller for what physical inputs a controller has for easier config

Reply:
>A key is unique, so we can't assign multiples. 

That's exactly the problem.

Reply:
>we do need more input types as we add more physical controllers with more buttons.

No we don't. A button is a button

Reply:
> No we don't. A button is a button

We currently have explicit `Menu` and `GripPress` (etc) buttons defined. Why are those not just "button"? How do you differentiate between different buttons in the controller mapping profile if they're all just `Button`?